osf_tests/management_commands/test_backfill_egap_metadata.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+
+from osf.management.commands.backfill_egap_provider_metadata import backfill_egap_metadata
+from osf.management.commands.update_registration_schemas import update_registration_schemas
+from osf.models import RegistrationSchema
+from osf_tests.factories import RegistrationFactory, RegistrationProviderFactory
+
+EGAP_ID_SCHEMA_KEY = 'q3'
+REGISTRATION_DATE_SCHEMA_KEY = 'q4'
+OLDER_EGAP_ID = 'ABC123'
+OLDER_TIMESTAMP = '2020-10-04 08:30:00 -0400'
+OLDEST_EGAP_ID = 'XYZ789'
+OLDEST_TIMESTAMP = '03/01/2011 - 22:00'
+
+
+@pytest.mark.django_db
+class TestMigrateEgapRegistrationMetadata:
+
+    @pytest.fixture()
+    def egap(self):
+        return RegistrationProviderFactory(_id='egap')
+
+    @pytest.fixture()
+    def older_registration(self, egap):
+        schema = RegistrationSchema.objects.get(name='EGAP Registration', schema_version=3)
+        registration = RegistrationFactory(schema=schema)
+        registration.provider = egap
+        registration.registration_responses[EGAP_ID_SCHEMA_KEY] = OLDER_EGAP_ID
+        registration.registration_responses[REGISTRATION_DATE_SCHEMA_KEY] = OLDER_TIMESTAMP
+        registration.save()
+        return registration
+
+    @pytest.fixture()
+    def oldest_registration(self, egap):
+        schema = RegistrationSchema.objects.get(name='EGAP Registration', schema_version=2)
+        registration = RegistrationFactory(schema=schema)
+        registration.provider = egap
+        registration.registration_responses[EGAP_ID_SCHEMA_KEY] = OLDEST_EGAP_ID
+        registration.registration_responses[REGISTRATION_DATE_SCHEMA_KEY] = OLDEST_TIMESTAMP
+        registration.save()
+        return registration
+
+    @pytest.fixture()
+    def newer_registration(self, egap):
+        try:
+            schema = RegistrationSchema.objects.get(name='EGAP Registration', schema_version=4)
+        except Exception:
+            update_registration_schemas()
+            schema = RegistrationSchema.objects.get(name='EGAP Registration', schema_version=4)
```

Comment:
this try/except may not be necessary on a fresh deployment based on the problems I had with the `test_schemas_list_crud` test above on GHA.